custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -354,9 +354,6 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
             vm.years = _.filter(vm.yearsCopy, function (y) {
                 return y.id >= 2018;
             });
-            vm.months = _.filter(vm.monthsCopy, function (month) {
```

Comment:
This appears to remove the part that hides the ineligible months. It will correctly default the date to a valid month but users will still be able to select an invalid one.

Reply:
the check has been moved to the inner method setMonthToPreviousIfAfterThe15thAndTwoMonthsIfBefore15th